### PR TITLE
Adding support for Hacro 8

### PR DESCRIPTION
**Pull request by Flatmasterflex**
## Description

Support for a new board designed by Short Circuits. The Hacro 8 is a 32u4 based board with ws2812's under each key. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* none

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hacro_8/hacro_8.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef HACRO8_H
+#define HACRO8_H
```
Comment (zvecr) [inline keyboards/hacro_8/hacro_8.h:2]:
```suggestion
#pragma once
```

File: keyboards/hacro_8/hacro_8.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#define LAYOUT( \
+      k00, k01, k02, k03, \
+      k04, k05, k06, k07  \
+) \
+{ \
+    { k00, k01, k02, k03 }, \
+    { k04, k05, k06, k07 } \
+} 
+
+#endif
```
Comment (zvecr) [inline keyboards/hacro_8/hacro_8.h:15]:
```suggestion
```

File: keyboards/hacro_8/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#include QMK_KEYBOARD_H
+
+#define KC_CAPW LGUI(LSFT(KC_3))        // Capture whole screen
+#define KC_CPYW LGUI(LSFT(LCTL(KC_3)))  // Copy whole screen
+#define KC_CAPP LGUI(LSFT(KC_4))        // Capture portion of screen
+#define KC_CPYP LGUI(LSFT(LCTL(KC_4)))  // Copy portion of screen
+#define KC_X0 LT(_FN, KC_ESC)
+
```
Comment (zvecr) [inline keyboards/hacro_8/keymaps/default/keymap.c:8]:
```suggestion

```

File: keyboards/hacro_8/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT(
+	    KC_Q, KC_W, KC_E, KC_R, \
```
Comment (zvecr) [inline keyboards/hacro_8/keymaps/default/keymap.c:11]:
```suggestion
	    KC_Q, KC_W, KC_E, KC_R,
```

File: keyboards/hacro_8/rules.mk [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+MOUSEKEY_ENABLE  = yes  # Mouse keys
+EXTRAKEY_ENABLE  = yes  # Audio control and System control
+CONSOLE_ENABLE   = no # Console for debug
+COMMAND_ENABLE   = yes # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE      = yes  # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no # Enable keyboard backlight functionality
+AUDIO_ENABLE     = no
+RGBLIGHT_ENABLE  = yes
+OLED_DRIVER_ENABLE = no
```
Comment (zvecr) [inline keyboards/hacro_8/rules.mk:27]:
```suggestion
```

File: keyboards/hacro_8/readme.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Hacro 8
+
+The Hacro 8 is a 2x4 mechanical keyboard with a WS2812B under each key and 2 on the underside of the PCB. It uses a 32u4 with the caterina bootloader. 
+
+![Hacro_8](https://lh3.googleusercontent.com/LNV81nz8awpURCpk4A7lmaNVFCsmUr5_6B89qjRqX6CINRyT_DtHuhOacxQhUmwG_yohhmFlp7uv9_winY--B47fbbK9FZU8lJVO4XMCdA6r870o2OsuiUWXWPJw_8Dqqr21ZvpTPg=w2400)
+![Hacro_8_PCB](https://lh3.googleusercontent.com/GP4x0EBqyVq7chv3KFF3O3yM9DodspuB1AkPtNVFsCw5458kvMqt_hVDdPlkAO88aDI6_0YyIneCESLbFnoHbDu7aWJVtY5qBIAf6m6MTNweYyVo30dJSXAs30CbDbWLKhaIvYNkoQ=w2400)
+
+* Keyboard Maintainer: Martyn Evans
+* Hardware Supported: Hacro 8
+* Hardware Availability: etsy link
```
Comment (zvecr) [inline keyboards/hacro_8/readme.md:10]: Maybe add a link here?

File: keyboards/hacro_8/keymaps/default/config.h [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (zvecr) [inline keyboards/hacro_8/keymaps/default/config.h:1]: This file could be deleted.

**Comment by zvecr** [comment]
Also a `info.json` should be added, info is available [here](https://docs.qmk.fm/#/reference_info_json) with a handy [kle converter](https://qmk.fm/converter/).

File: keyboards/hacro_8/readme.md (hunk 6 above)
Comment (Flatmasterflex) [reply keyboards/hacro_8/readme.md:10]: No link available at this time. I've moved on to a different project, so I'll have to update this when I get back to working on the keyboard. Cheers

File: keyboards/hacro_8/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#include QMK_KEYBOARD_H
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT(
+	    KC_Q, KC_W, KC_E, KC_R,
+		KC_A, KC_S, KC_D, KC_F  
+	)
+};
```
Comment (fauxpark) [inline keyboards/hacro_8/keymaps/default/keymap.c:7]:
```suggestion

const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    [0] = LAYOUT(
        KC_Q, KC_W, KC_E, KC_R,
        KC_A, KC_S, KC_D, KC_F
    )
};
```

File: keyboards/hacro_8/rules.mk [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/hacro_8/rules.mk:11]:
```suggestion
```

File: keyboards/hacro_8/rules.mk [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   comment out to disable the options.
```
Comment (fauxpark) [inline keyboards/hacro_8/rules.mk:15]:
```suggestion
#   change yes to no to disable
```

File: keyboards/hacro_8/rules.mk [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+BOOTMAGIC_ENABLE = yes  # Virtual DIP switch configuration
+MOUSEKEY_ENABLE  = yes  # Mouse keys
+EXTRAKEY_ENABLE  = yes  # Audio control and System control
+CONSOLE_ENABLE   = no # Console for debug
+COMMAND_ENABLE   = yes # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE      = yes  # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no # Enable keyboard backlight functionality
+AUDIO_ENABLE     = no
+RGBLIGHT_ENABLE  = yes
```
Comment (fauxpark) [inline keyboards/hacro_8/rules.mk:26]:
```suggestion
BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/hacro_8/config.h [hunk 12]
```diff
@@ -0,0 +1,37 @@
+#pragma once
```
Comment (drashna) [inline keyboards/hacro_8/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/hacro_8/hacro_8.c [hunk 13]
```diff
@@ -0,0 +1 @@
+#include "hacro_8.h"
```
Comment (drashna) [inline keyboards/hacro_8/hacro_8.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/hacro_8/hacro_8.h [hunk 14]
```diff
@@ -0,0 +1,14 @@
+#pragma once
```
Comment (drashna) [inline keyboards/hacro_8/hacro_8.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/hacro_8/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,8 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/hacro_8/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**